ansible/files/home/deployer/webhook/lib/deployer/app.rb

```diff
@@ -13,10 +14,38 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+require "openssl"
+require_relative "response"
+
 module Deployer
   class App
     def call(env)
-      [200, {}, ["Hello deployer"]]
+      request = Rack::Request.new(env)
+      response = Response.new
+      process(request, response) or response.finish
+    end
+
+    private
+
+    def process(request, response)
+      unless request.post?
+        response.set(:method_not_allowed, "must POST")
+        return nil
+      end
+
+      unless verify_signature(request)
+        response.set(:unauthorized, "Authorization failed")
+        return nil
+      end
+
+      response.finish
+    end
+
+    def verify_signature(request)
+      signature = 'sha256=' + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'),
+                                                      ENV['SECRET_TOKEN'],
+                                                      request.body.read)
```

Comment:
変数をもう一つ追加したほうが読みやすいんじゃないかな。

```suggestion
      hmac_sha256 = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'),
                                            ENV['SECRET_TOKEN'],
                                            request.body.read)
      signature = "sha256=#{hmac}"
```

Reply:
fix: [53df47d](https://github.com/groonga/packages.groonga.org/pull/46/commits/53df47d3364f8575b9187eb15a4165a2b2da3618)